packages/block-library/src/navigation/index.php

```diff
@@ -204,8 +205,16 @@ function core_block_navigation_build_html( $attributes, $block, $colors, $font_s
 
 	foreach ( (array) $block['innerBlocks'] as $key => $block ) {
 		$has_submenu = count( (array) $block['innerBlocks'] ) > 0;
+		$is_active   = ! empty( $block['attrs']['id'] ) && ( get_the_ID() === $block['attrs']['id'] );
+
+		$class_name = ! empty( $block['attrs']['className'] ) ? implode( ' ', (array) $block['attrs']['className'] ) : false;
+
+		if ( false !== $class_name ) {
+			$css_classes .= ' ' . $class_name;
```

Comment:
`$css_classes` was declared outside of the loop, so each run is appending whatever custom class that block might have, with the result that custom classes are being added to all subsequent nav items:

<img width="642" alt="Screen Shot 2020-03-18 at 3 10 56 pm" src="https://user-images.githubusercontent.com/8096000/76924633-1381b800-692b-11ea-8582-88d3cc3273ed.png">

Reply:
Thank you! Good catch, fixed it!